marlowe-playground-client/src/JavascriptEditor/State.purs

```diff
@@ -85,8 +116,85 @@ handleAction _ SendResultToBlockly = do
 editorResize :: forall state action msg m. HalogenM state action ChildSlots msg m Unit
 editorResize = void $ query _jsEditorSlot unit (Monaco.Resize unit)
 
-editorSetValue :: forall state action msg m. String -> HalogenM state action ChildSlots msg m Unit
-editorSetValue contents = void $ query _jsEditorSlot unit (Monaco.SetText contents unit)
+decorationHeader :: Array String
+decorationHeader =
+  [ "import {"
+  , "    PK, Role, Account, Party, ada, AvailableMoney, Constant, NegValue, AddValue,"
+  , "    SubValue, MulValue, Scale, ChoiceValue, SlotIntervalStart, SlotIntervalEnd,"
+  , "    UseValue, Cond, AndObs, OrObs, NotObs, ChoseSomething, ValueGE, ValueGT,"
+  , "    ValueLT, ValueLE, ValueEQ, TrueObs, FalseObs, Deposit, Choice, Notify,"
+  , "    Close, Pay, If, When, Let, Assert, SomeNumber, AccountId, ChoiceId, Token,"
+  , "    ValueId, Value, EValue, Observation, Bound, Action, Payee, Case, Contract"
+  , "} from 'marlowe-js';"
+  , ""
+  , "(function (): Contract {"
+  ]
+
+decorationFooter :: Array String
+decorationFooter = [ "})" ]
+
+decorationHeaderString :: String
+decorationHeaderString = joinWith "\n" decorationHeader
+
+lengthOfHeader :: Int
+lengthOfHeader = length decorationHeaderString
+
+decorationFooterString :: String
+decorationFooterString = joinWith "\n" decorationFooter
+
+lengthOfFooter :: Int
+lengthOfFooter = length decorationFooterString
 
-editorGetValue :: forall state action msg m. HalogenM state action ChildSlots msg m (Maybe String)
-editorGetValue = query _jsEditorSlot unit (Monaco.GetText identity)
+editorSetValue ::
+  forall m.
+  MonadAff m =>
+  String ->
+  HalogenM State Action ChildSlots Void m Unit
+editorSetValue contents = do
+  let
+    decoratedContent = joinWith "\n" [ decorationHeaderString, contents, decorationFooterString ]
+
+    numLines = Array.length $ lines decoratedContent
+  void $ query _jsEditorSlot unit $ Monaco.SetText decoratedContent unit
+  mTopDecorationId <- query _jsEditorSlot unit $ Monaco.SetDeltaDecorations 1 (Array.length decorationHeader) identity
+  mBottomDecorationId <- query _jsEditorSlot unit $ Monaco.SetDeltaDecorations (numLines - Array.length decorationFooter + 1) numLines identity
+  for_ mTopDecorationId
```

Comment:
I think using `MaybeT` would make this a lot more readable, I think it would be something like:
```purescript
void $ runMaybeT do
  topDecorationId <- mTopDecorationId
  bottomDecorationId <- mBottomDecorationId
  lift $ assign _decorationIds { topDecorationId, bottomDecorationId }
```